src/filter/mod.rs

```diff
@@ -538,23 +539,26 @@ fn unapply2<'a>(
         }
         Op::Workspace(path) => {
             let root = to_filter(Op::Subdir(path.to_owned()));
-            let mapped = parse(&tree::get_blob(
-                &transaction.repo(),
-                &tree,
-                &Path::new("workspace.josh"),
-            ))?;
+            let mapped = &tree::get_blob(&transaction.repo(), &tree, &Path::new("workspace.josh"));
+            let parsed = parse(mapped)?;
+
+            let mut blob = String::new();
+            if let Ok(c) = get_comments(mapped) {
+                if c.len() > 0 {
+                    blob = format!("{}", c);
+                }
+            }
+            let blob = &format!("{}{}\n", &blob, pretty(parsed, 0));
```

Comment:
Will there allways be a newline at the end of the comments, or would this better be `{}\n{\n}\n`?

Reply:
There will, because of the grammar, I think.

Reply:
Also all the newlines are kept as part of the comments. If we add one here, each push will increase the number of newlines by one.